repair/row_level.cc

```diff
@@ -594,13 +594,20 @@ future<std::list<repair_row>> to_repair_rows_list(repair_rows_on_wire rows, sche
     lw_shared_ptr<const decorated_key_with_hash> dk_ptr;
     lw_shared_ptr<mutation_fragment> last_mf;
     position_in_partition::tri_compare cmp(*s);
-    for (partition_key_and_mutation_fragments& x : rows) {
+    for (auto it = rows.begin(); it != rows.end(); ) {
+        auto x = std::move(*it);
+        it = rows.erase(it);
```

Comment:
Why not put this in the `for` itself? The increment part happens after the loop body is executed.

Reply:
I think it's more efficient this way with respect to the cpu dcache, otherwise we need to refetch the iterator value at the end of the loop just for erasing it.
It would be nice to use a generator to "stream" the list in the for loop...

Reply:
although, that said, we need to refetch it for the end condition as well.  maybe it won't make a difference.

Reply:
another option is
```c++
while (!rows.empty()) {
    auto it = rows.begin();
    auto x = std::move(*it);
    rows.erase(it);
}
```

Reply:
I think it is safe to go for readability here, the reader's main memory is much slower than that of the machine's.